src/training/tech.rs

```diff
@@ -1,278 +1,269 @@
+        module_accessor,
+        *FIGHTER_STATUS_TRANSITION_TERM_ID_PASSIVE,
+    ) && (second_prev_status != FIGHTER_STATUS_KIND_CATCHED_AIR_FALL_GANON);
+
+    if !can_tech {
+        return false;
+    }
+
+    let do_tech: bool = match state {
+        TechFlags::IN_PLACE => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE.as_lua_int();
+            *unk = LUA_TRUE;
+            true
+        }
+        TechFlags::ROLL_F => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE_FB.as_lua_int();
+            *unk = LUA_TRUE;
+            TECH_ROLL_DIRECTION = Direction::IN; // = In
+            true
+        }
+        TechFlags::ROLL_B => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE_FB.as_lua_int();
+            *unk = LUA_TRUE;
+            TECH_ROLL_DIRECTION = Direction::OUT; // = Away
+            true
+        }
+        _ => false,
+    };
+    if do_tech && MENU.mash_triggers.contains(MashTrigger::TECH) {
+        mash::buffer_menu_mash();
+    }
+
+    true
+}
+
+unsafe fn handle_wall_tech(
+    module_accessor: &mut app::BattleObjectModuleAccessor,
+    status_kind: &mut L2CValue,
+    unk: &mut L2CValue,
+    status_kind_int: i32,
+    state: TechFlags,
+) -> bool {
+    let can_tech = WorkModule::is_enable_transition_term(
+        module_accessor,
+        *FIGHTER_STATUS_TRANSITION_TERM_ID_PASSIVE_WALL,
+    );
+
+    if ![
+        *FIGHTER_STATUS_KIND_STOP_WALL,
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_REFLECT_LR,
+    ]
+    .contains(&status_kind_int)
+        || state == TechFlags::NO_TECH
+        || !can_tech
+    {
+        return false;
+    }
+
+    let do_tech: bool = match state {
+        TechFlags::IN_PLACE => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE_WALL.as_lua_int();
+            *unk = LUA_TRUE;
+            true
+        }
+        TechFlags::ROLL_F => {
+            *status_kind = FIGHTER_STATUS_KIND_PASSIVE_WALL_JUMP.as_lua_int();
+            *unk = LUA_TRUE;
+            true
+        }
+        _ => false,
+    };
+    if do_tech && MENU.mash_triggers.contains(MashTrigger::TECH) {
+        mash::buffer_menu_mash();
+    }
+    true
+}
+
+unsafe fn handle_ceil_tech(
+    module_accessor: &mut app::BattleObjectModuleAccessor,
+    status_kind: &mut L2CValue,
+    unk: &mut L2CValue,
+    status_kind_int: i32,
+    state: TechFlags,
+) -> bool {
+    let can_tech = WorkModule::is_enable_transition_term(
+        module_accessor,
+        *FIGHTER_STATUS_TRANSITION_TERM_ID_PASSIVE_CEIL,
+    );
+
+    if ![
+        *FIGHTER_STATUS_KIND_STOP_CEIL,
+        *FIGHTER_STATUS_KIND_DAMAGE_FLY_REFLECT_U,
+    ]
+    .contains(&status_kind_int)
+        || state == TechFlags::NO_TECH
+        || !can_tech
+    {
+        return false;
+    }
+
+    *status_kind = FIGHTER_STATUS_KIND_PASSIVE_CEIL.as_lua_int();
+    *unk = LUA_TRUE;
+    if MENU.mash_triggers.contains(MashTrigger::TECH) {
+        mash::buffer_menu_mash();
+    }
+    true
```

Comment:
Same

Reply:
See earlier comment. Note that we didn't need a `do_tech` here because we didn't need to account for more than one tech option.